### PR TITLE
WELD-2484 Configurators should override toString()

### PR DESCRIPTION
**Pull request by manovotn**
Not sure this is 'sensible enough'. It can blow up (and call toString()) from many scenarios so amount of useful information may vary.

### CODE REVIEW

File: impl/src/main/java/org/jboss/weld/bootstrap/events/configurator/InjectionPointConfiguratorImpl.java [hunk 1]
```diff
@@ -210,6 +210,10 @@ public boolean isTransient() {
            return isTransient;
        }
 
+       @Override
+       public String toString() {
+           return "InjectionPoint of type " + requiredType + " with qualifiers " + qualifiers +
+               " which is " + (isDelegate ? "" : "not ") + "delegate and is" + (isTransient ? "" : "not ") + "transient.";
```
Comment (mkouba) [inline impl/src/main/java/org/jboss/weld/bootstrap/events/configurator/InjectionPointConfiguratorImpl.java]: `delegate=true` style would be just fine.

**Comment by mkouba** [comment]
I wonder if we should add sensible `toString()` to all products of all configurators. Some of them already have it...

**Comment by manovotn** [comment]
Hmm, I think only `BeanConfigurator` had this altered so far.

File: impl/src/main/java/org/jboss/weld/bootstrap/events/configurator/ObserverMethodConfiguratorImpl.java [hunk 2]
```diff
@@ -330,6 +330,14 @@ public boolean isEventMetadataRequired() {
            return true;
        }
 
-   }
+       @Override
+       public String toString() {
+           return "Configurator observer method [Bean class = " + getBeanClass().toString()
```
Comment (mkouba) [inline impl/src/main/java/org/jboss/weld/bootstrap/events/configurator/ObserverMethodConfiguratorImpl.java]: I'm not sure why but we allow to set `null` bean class. If you remove `toString()` here, then this method should not throw NPE and the resulting string should be the same.

Comment (manovotn) [reply impl/src/main/java/org/jboss/weld/bootstrap/events/configurator/ObserverMethodConfiguratorImpl.java]: Nice catch!

**Pull request merged**